4_naming_conventions/42_directory_names.md

```diff
@@ -0,0 +1,154 @@
+     documentation and/or other materials provided with the distribution.
+
+  THIS DOCUMENTATION IS PROVIDED BY TIANOCORE PROJECT "AS IS" AND ANY EXPRESS OR
+  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+  EVENT SHALL TIANOCORE PROJECT  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS DOCUMENTATION, EVEN IF
+  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+-->
+
+## 4.2 Directory Names
+Below sections are the directory naming guidelines for EDK II modules. The guidelines are not
+just considering the the uniformity of directory naming, but it also provides the flexibility of
+directory name construction for the scenario of different EDK II module designs; such as the
+support for multiple processor architectures and vendors. It may require the further discussions
+between EDK II maintainers and contributors in order to determine the best naming of the EDK II
+module directory.
+
+#### 4.2.1 EDKII package directory
+
+```
+<PackageName>Pkg
+
+   <PackageName> REQUIRED  *
+```
+
+#### 4.2.2 EDKII Module directory
+
+* If the implementation is general for all CPU architectures without the specific arch dependency:
+```
+<Feature><Phase>
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm, Smm,
+                              Uefi.
+
+Example:
+   SmbiosDxe/
+```
+
+* If the implementation is for the specific CPU architectures, CPU vendors or the implementation is shared by certain CPU archs:
+```
+<Feature><Phase><CpuArch><Vendor>
+  or
+<Feature><Phase>/<CpuArch>/<Vendor>
+
+   <Feature>      REQUIRED    * 
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                              StandaloneMm, Smm, Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'   
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+                              
+   <Vendor>       OPTIONAL    *
+                              
+Example:
+   - CpuDxe/                    # First implementation of CpuDxe.
+   - CpuDxe/RiscV64/            # RiscV64 specific implementation.
+           /                    # Common files for the RiscV64 and other archs.
+   - CpuDxeRiscv64/             # RiscV64 implementation of CpuDxe.

```

Comment:
In which case this will be required? Isn't generic module and arch specific folder inside sufficient always? For ex: Even if the module is created fresh for RISC-V, can we not name it generic since we don't know in future some other architecture may need it?

Reply:
Those are all examples we can take for the module. Usually, if the module is only for the specific arch for sure then we can choose <Feature>Riscv64 (e,g, PlicDxeRiscV64). You can have CpuDxe/RiscV64 if you consider this module would have other archs. Or just use the generic name like CpuDxe. However, other archs may need a reconstruction later.  Conversations between contributor and maintainer may be required for the final decision.
Please also check the Wiki page, I added you as the collaborator,
[The Guidelines of Reconstruct EDK II Modules for Processor Architectures and Vendors' Implementation](https://github.com/changab/tianocore.github.io/wiki/The-Guidelines-of-Reconstruct-EDK-II-Modules-for-Processor-Architectures-and-Vendors'-Implementation)

Reply:
Okay. Thanks!